dashboard/modules/snapshot/snapshot_head.py

```diff
@@ -89,6 +98,41 @@ async def snapshot(self, req):
             success=True, message="hello", snapshot=snapshot
         )
 
+    @routes.get("/api/component_activities")
+    async def get_component_activities(self, req) -> aiohttp.web.Response:
+        # Get activity information for driver
+        driver_activity_info = await self._get_job_activity_info()
+
+        resp = {"driver": dataclasses.asdict(driver_activity_info)}
+        return dashboard_optional_utils.rest_response(
+            success=True,
+            convert_google_style=False,
+            ray_activity_response=resp,
+        )
+
+    async def _get_job_activity_info(self) -> RayActivityResponse:
+        # Returns if there is Ray activity from drivers (job).
+        # Drivers in namespaces that start with _ray_internal are not
+        # considered activity.
+        request = gcs_service_pb2.GetAllJobInfoRequest()
+        reply = await self._gcs_job_info_stub.GetAllJobInfo(request, timeout=5)
```

Comment:
Can you make this timeout configurable? (and we can make it equal to the ping frequency)